custom/plugins/ProductExtension/src/Services/ProductCustomFieldService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProductExtension\Services;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\System\CustomField\CustomFieldTypes;
+
+class ProductCustomFieldService
+{
+    public const FIELD_SET_NAME = 'custom_product_fields';
+    public const FIELD_NAME = 'product_history';
+
+    private EntityRepository $customFieldSetRepository;
+
+    public function __construct(EntityRepository $customFieldSetRepository)
+    {
+        $this->customFieldSetRepository = $customFieldSetRepository;
+    }
+
+    public function findProductCustomFieldSetIds(Context $context): array
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('name', self::FIELD_SET_NAME));
+
+        return $this->customFieldSetRepository->searchIds($criteria, $context)->getIds();
+    }
+
+    public function createProductCustomFieldSet(Context $context)
```

Comment:
Tutaj nie byłem pewny co to znaczy, aktywacja, deaktywacja, więc wykonałem to poprostu w ten sposób, że na aktywację dodaje konkretne pole